handlers/workspaces_test.go

```diff
@@ -930,7 +931,90 @@ func TestGetWorkspaceUsers(t *testing.T) {
 }
 
 func TestGetUserDropdownWorkspaces(t *testing.T) {
+	teardownSuite := SetupSuite(t)
+	defer teardownSuite(t)
+
+	db.TestDB.DeleteWorkSpaceAllData()
+
+	oHandler := NewWorkspaceHandler(db.TestDB)
+
+	person := db.Person{
+		Uuid:        uuid.New().String(),
+		OwnerAlias:  "test-alias",
+		UniqueName:  "test-unique-name",
+		OwnerPubKey: "test-pubkey",
+		PriceToMeet: 0,
+		Description: "test-description",
+	}
+	person2 := db.Person{
+		Uuid:        uuid.New().String(),
+		OwnerAlias:  "test-alias2",
+		UniqueName:  "test-unique-name2",
+		OwnerPubKey: "test-pubkey2",
+		PriceToMeet: 0,
+		Description: "test-description2",
+	}
+	db.TestDB.CreateOrEditPerson(person)
+	db.TestDB.CreateOrEditPerson(person2)
+
+	workspace := db.Workspace{
+		Uuid:        uuid.New().String(),
+		Name:        "test-workspace" + uuid.New().String(),
+		OwnerPubKey: person2.OwnerPubKey,
+		Github:      "https://github.com/test",
+		Website:     "https://www.testwebsite.com",
+		Description: "test-description",
+	}
+	db.TestDB.CreateOrEditWorkspace(workspace)
+	workspace = db.TestDB.GetWorkspaceByUuid(workspace.Uuid)
+	ctx := context.WithValue(context.Background(), auth.ContextKey, workspace.OwnerPubKey)
+
+	roles := []db.WorkspaceUserRoles{
+		{WorkspaceUuid: workspace.Uuid, OwnerPubKey: person2.OwnerPubKey, Role: "ADD BOUNTY"},
+		{WorkspaceUuid: workspace.Uuid, OwnerPubKey: person2.OwnerPubKey, Role: "UPDATE BOUNTY"},
+		{WorkspaceUuid: workspace.Uuid, OwnerPubKey: person2.OwnerPubKey, Role: "DELETE BOUNTY"},
+		{WorkspaceUuid: workspace.Uuid, OwnerPubKey: person2.OwnerPubKey, Role: "PAY BOUNTY"},
+	}
+	db.TestDB.CreateUserRoles(roles, workspace.Uuid, person2.OwnerPubKey)
+
+	dbPerson := db.TestDB.GetPersonByUuid(person2.Uuid)
+
+	handlerUserHasAccess := func(pubKeyFromAuth string, uuid string, role string) bool {
```

Comment:
@MuhammadUmer44 You should not mock this, the HasAccess Status should come from the controller/ business logic, mocking this defeats the purpose of this test.

Reply:
@elraphty I check it